patchwork/steps/PR/PR.py

```diff
@@ -2,21 +2,50 @@
 from patchwork.step import Step
 from patchwork.steps.CommitChanges.CommitChanges import CommitChanges
 from patchwork.steps.CreatePR.CreatePR import CreatePR
-from patchwork.steps.PR.typed import PRInputs
+from patchwork.steps.PR.typed import PRInputs, PROutputs
 from patchwork.steps.PreparePR.PreparePR import PreparePR
 
 
-class PR(Step):
-    required_keys = PRInputs.__required_keys__
+class PR(Step, input_class=PRInputs, output_class=PROutputs):
 
     def __init__(self, inputs):
         super().__init__(inputs)
-        missing_keys = self.required_keys.difference(set(inputs.keys()))
-        if len(missing_keys) > 0:
-            raise ValueError(f'Missing required data: "{missing_keys}"')
-
         self.inputs = inputs
-        return
+        self.__handle_modified_code_files()
+
+
+    def __handle_modified_code_files(self):
+        if self.inputs.get("modified_code_files") is not None:
```

Comment:
If I understand correctly, we don't proceed if `modified_code_files` is not none and we simply want to handle `modified_files` if `modified_files` is none? Just checking if `modified_files` is deliberate here. 

Otherwise looks good!

Reply:
This is more of to redirect `modifed_files` to `modified_code_files` if `modified_code_files` is not present.